aes/src/armv8/expand.rs

```diff
@@ -37,9 +41,9 @@ pub(super) fn expand_key<const L: usize, const N: usize>(key: &[u8; L]) -> [uint
         let mut word = ek_words[i - 1];
 
         if i % nk == 0 {
-            word = sub_word(word).rotate_right(8) ^ ROUND_CONSTS[i / nk - 1];
+            word = unsafe { sub_word(word) }.rotate_right(8) ^ ROUND_CONSTS[i / nk - 1];
         } else if nk > 6 && i % nk == 4 {
-            word = sub_word(word)
+            word = unsafe { sub_word(word) };
```

Comment:
Since `expand_key` is not marked as `#[target_feature(enable = "aes")]`, I think these `sub_word` calls would not be inlined if `aes` target feature is not enabled at compile time. I wonder if that may potentially affect performance.

Reply:
Yeah, I could just throw `#[target_feature(enable = "aes")]` on it and see how it impacts performance.

The performance is already something on the order of something like 80ns though, and this is just the initial key setup, so it's something of a micro-optimization. I can see what the impact is.

Reply:
I tried it and it was 64ns with or without (noise of +/-2). It's a change I might make if I refactor that method anyway though.

Reply:
On which platform did you benchmark?

On aarch64 macOS, `aes` target feature is enabled at compile time by default so the compiler can inline all `#[target_feature(enable = "aes")]` functions regardless of whether `#[target_feature(enable = "aes")]` is used by the caller. The aes feature must be disabled at compile time to measure performance regarding the optimization I'm discussing.


In any case, I agree with your opinion that this should not significantly improve performance, because it is not called in a hot path.

Reply:
Aah yes, it was on macOS